Rearrange the docs (#144)
kyma-project/kyma-environment-broker: docs/user/05-50-configure-list-of-modules.md

# Configure list of modules

> By default, Kyma Environment Broker applies the Kyma CR, including the default modules selected by Kyma, to a cluster.

Reviewer: ```suggestion
By default, Kyma Environment Broker applies the Kyma custom resource, including the default modules selected by Kyma, to a cluster.
```

> You can configure your custom list of modules by setting the "modules" object in the JSON schema.

Reviewer: ```suggestion
You can configure your custom list of modules by setting the **modules** object in the JSON schema.
```
I'm not sure about this, but I guess we could use bold font for "modules" and "oneof". I think they're both JSON Schema keywords (for example, see this [doc](https://json-schema.org/blog/posts/applicability-json-schema-fundamentals-part-1#putting-everything-together-avoiding-oneof-pitfalls))

Reviewer: Thank you for clarifying that

To do that, use the following two fields:
- **default** (bool) - defines whether to use the default list of modules
- **list** (array) - defines a custom list of modules

> API for module configuration is built on the "oneOf" feature from the JSON schema. If the "modules" object is passed to API, it must have only one valid option. Thus, to pass JSON API Validator, you must set only one field. See examples below.

Reviewer: ```suggestion
API for module configuration is built on the **oneOf** feature from the JSON schema. If the **modules** object is passed to API, it must have only one valid option. Thus, to pass JSON API Validator, you must set only one field. See examples below.
```

## Correct API calls

- In the default mode, the **modules** object is set to `default: true` and modules are pre-selected by Kyma. The same happens when the **modules** section is not set at all (mapped to nil) in the payload.

   ```
   "modules": {
       "default": true
   }
   ```

- If you want to use your custom modules' list configuration, you must pass the **list** with the modules you want installed.

   ```
   "modules": {
       "list": [
           {
               "name": "btp-operator",
               "customResourcePolicy": "CreateAndDelete"
           },
           {
               "name": "keda",
               "channel": "fast",
           }
       ]
   }
   ```

- The following calls result in Kyma runtime installation without any modules.

   ```
   "modules": {
       "list": []
   }
   ```

   ```
   "modules": {
       "default": false
   }
   ```

## Incorrect API calls

> - A call with the **modules** section empty fails due to using the "oneOf" feature in the JSON schema.

Reviewer: ```suggestion
- A call with the **modules** section empty fails due to using the **oneOf** feature in the JSON schema.
```

   ```
   "modules": {}
   ```

> - A call with the **modules** section with both parameters filled in fails due to using the "oneOf" feature in the JSON schema.

Reviewer: ```suggestion
- A call with the **modules** section with both parameters filled in fails due to using the **oneOf** feature in the JSON schema.
```
